lahja/endpoint.py

```diff
@@ -162,6 +170,23 @@ def event_loop(self) -> asyncio.AbstractEventLoop:
 
         return self._loop
 
+    def check_event_loop(func: TFunc) -> TFunc:  # type: ignore
```

Comment:
Do I read that right that this means the `Endpoint` can not run with a different event loop than the default event loop? This may get into trouble with the Trinity tests where we create an event loop explicitly.

https://github.com/ethereum/trinity/blob/b7eaa8a195aecbacfd982d03adfde976ded6298f/tests/conftest.py#L105-L143

Reply:
Ok, looking at your Trinity PR, it looks like we get around manually setting event loops. I'm ok going down that way :+1:

Reply:
Yeah, the change to trinity to allow this was surprisingly small!

It's still possible to use lahja on a non-default event loop. Lahja uses the event loop which is active the first time one of its async methods are called. The code is simpler but this drops the ability to tell lahja to use an event loop which has not yet been entered.